test/test_sklearn_compat.py

```diff
@@ -110,6 +112,54 @@ def generate_array_like(input_data, labels=None):
   return input_data_changed, labels_changed
 
 
+# TODO: Find a better way to run this test and the next one, to avoid
+# duplicated code.
```

Comment:
does this TODO still holds, or you have precisely done that?

Reply:
I can use `isinstance()` and join the tests `test_array_like_inputs_mahalanobis` and `test_array_like_inputs` that is just next to the first.

or

Have these two tests separately, which I dont like because there is duplicated code.

Take `test_deprecated_score_pairs_same_result` at `test_base_metric.py` as an example test using `isinstance()`